### PR TITLE
Startup race condition

### PR DESCRIPTION
**Pull request by tmat**
dotnet watch has a rather complex start-up routine that creates multiple tasks and awaits them at various times. 
Some of these tasks were not awaited in the right order, specifically the solution might have been loaded by MSBuildWorkspace only after we captured its content as a baseline for Hot Reload session. The EnC analyzer then wasn't able to determine the correct set of changes. The test added in this PR was failing randomly.

### CODE REVIEW

File: src/BuiltInTools/dotnet-watch/Properties/launchSettings.json [hunk 1]
```diff
@@ -3,10 +3,10 @@
     "dotnet-watch": {
       "commandName": "Project",
       "commandLineArgs": "--verbose",
-      "workingDirectory": "$(RepoRoot)src\\Assets\\TestProjects\\BlazorWasmWithLibrary\\blazorwasm",
+      "workingDirectory": "D:\\Temp\\fs\\App",
```
Comment (tlmii) [inline src/BuiltInTools/dotnet-watch/Properties/launchSettings.json:6]: Intentional change?

File: src/Tests/dotnet-watch.Tests/Watch/Utilities/WatchableApp.cs [hunk 2]
```diff
@@ -121,7 +121,7 @@ public void Start(string projectRootPath, IEnumerable<string> arguments, string
             TestFlags testFlags = TestFlags.RunningAsTest,
             [CallerMemberName] string name = null)
         {
-            var args = new[] { "run", "--" };
+            var args = new[] { "run", "--verbose", "--" };
```
Comment (tlmii) [inline src/Tests/dotnet-watch.Tests/Watch/Utilities/WatchableApp.cs:124]: Did you want this change committed? I'm not familiar yet on what the output would do to the test run logs.

File: src/BuiltInTools/dotnet-watch/Properties/launchSettings.json (hunk 1 above)
Comment (tmat) [reply src/BuiltInTools/dotnet-watch/Properties/launchSettings.json:6]: Ah, no.

**Review by tlmii — APPROVED** [review]

File: src/Assets/TestProjects/WatchAppWithProjectDeps/AppWithDeps/Program.cs [hunk 3]
```diff
@@ -16,7 +16,12 @@ public static void Main(string[] args)
             Console.WriteLine("Started");
             // Process ID is insufficient because PID's may be reused.
             Console.WriteLine($"Process identifier = {Process.GetCurrentProcess().Id}, {Process.GetCurrentProcess().StartTime:hh:mm:ss.FF}");
-            Thread.Sleep(Timeout.Infinite);
+
+            while (true)
+            {
+                Lib.Print();
```
Comment (CyrusNajmabadi) [inline src/Assets/TestProjects/WatchAppWithProjectDeps/AppWithDeps/Program.cs:22]: out of curiosity, what does this Print statement do?  (maybe doc?)

Comment (tmat) [reply src/Assets/TestProjects/WatchAppWithProjectDeps/AppWithDeps/Program.cs:22]: This is a test project. Lib is F# class with method Print on it.

Comment (CyrusNajmabadi) [reply src/Assets/TestProjects/WatchAppWithProjectDeps/AppWithDeps/Program.cs:22]: oh, sorry :)

**Pull request merged**